src/main/java/com/example/onlinebookstore/controller/OrderController.java

```diff
@@ -0,0 +1,67 @@
+package com.example.onlinebookstore.controller;
+
+import com.example.onlinebookstore.dto.OrderItemResponseDto;
+import com.example.onlinebookstore.dto.OrderRequestDto;
+import com.example.onlinebookstore.dto.OrderResponseDto;
+import com.example.onlinebookstore.dto.OrderStatusUpdateDto;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.service.OrderService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Order management", description = "Endpoints for managing orders")
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/orders")
+public class OrderController {
+
+    private final OrderService orderService;
+
+    @PostMapping
+    @Operation(summary = "Place an order")
+    @ResponseStatus(HttpStatus.CREATED)
+    public OrderResponseDto placeOrder(@AuthenticationPrincipal User user,
+                                       @RequestBody OrderRequestDto orderRequestDto) {
```

Comment:
```suggestion
                                       @RequestBody @Valid OrderRequestDto orderRequestDto) {
```
Same for any other request dto's, we should always validate them

Reply:
done